tests/components/broadlink/test_device.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for Broadlink devices."""
+import broadlink.exceptions as blke
+import pytest
+
+from homeassistant.components.broadlink.const import DOMAIN
+from homeassistant.components.broadlink.device import BroadlinkDevice
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.setup import async_setup_component
+
+from . import pick_device
+
+from tests.async_mock import AsyncMock, MagicMock, patch
+
+
+async def test_device_setup(hass):
+    """Test a successful setup."""
+    device = pick_device(1)
+    mock_api = device.get_mock_api()
+    mock_entry = device.get_mock_entry()
+    device_entry = BroadlinkDevice(hass, mock_entry)
+    await async_setup_component(hass, DOMAIN, {})
+
+    with patch("broadlink.gendevice", return_value=mock_api), patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as mock_forward:
+        assert await device_entry.async_setup() is True
+
+    assert device_entry.api is mock_api
```

Comment:
Remove this.
```suggestion
```

Reply:
This is testing [this line](https://github.com/home-assistant/core/blob/9b570038d30cb794f35751ba0a96dcc04f38ca91/homeassistant/components/broadlink/device.py#L89).

Reply:
That's an integration detail that we should not test. It might change in the future and we don't want our tests stop that.

Reply:
Ok.

Reply:
Done.